core/node/state_keeper/src/io/mempool.rs

```diff
@@ -203,6 +203,13 @@ impl StateKeeperIO for MempoolIO {
                 continue;
             }
 
+            self.pool
+                .connection()
+                .await?
+                .blocks_dal()
+                .insert_l1_batch(cursor.l1_batch, timestamp, self.filter.fee_input)
```

Comment:
What if batch is inserted and right after that server restarts? I think you should check for batch with `is_sealed = false` at the beginning of this function. Perhaps you should also save `protocol_version`, `operator_address`

Reply:
Alright I just pushed dirty code that should do the trick. Still need to add a test for this scenario but I need to temporarily prioritize devx work so will get back to this PR next week.

As a side effect I had to bring `fee_address` back to the `l1_batches` table thus reverting some of your changes in https://github.com/matter-labs/zksync-era/pull/1617. Hope that's okay.

Reply:
looks good! waiting for test

Reply:
I have just added a test, PTAL